src/System.Diagnostics.Process/tests/ProcessTests.Unix.cs

```diff
@@ -62,12 +64,157 @@ public void TestRootGetProcessById()
+                    Assert.Equal(2, p.ExitCode);
+                }
+            }
+            else
+            {
+                Console.WriteLine($"{nameof(ProcessStart_UseShellExecuteTrue_OpenMissingFile_XdgOpenReturnsExitCode2)}(): {s_xdg_open} is not installed on this machine.");
+            }
+        }
+
+        [Theory, InlineData("/usr/bin/open"), InlineData("/usr/bin/nano")]
+        [PlatformSpecific(TestPlatforms.OSX)]
+        // [OuterLoop("Opens program")]
+        public void ProcessStart_OpenFileOnOsx_UsesSpecifiedProgram(string programToOpenWith)
+        {
+            string fileToOpen = GetTestFilePath() + ".txt";
+            File.WriteAllText(fileToOpen, $"{nameof(ProcessStart_OpenFileOnOsx_UsesSpecifiedProgram)}");
+            using (var px = Process.Start(programToOpenWith, fileToOpen))
+            {
+                Console.WriteLine($"in OSX, {nameof(programToOpenWith)} is {programToOpenWith}, while {nameof(px.ProcessName)} is {px.ProcessName}.");
+                // Assert.Equal(programToOpenWith, px.ProcessName); // on OSX, process name is dotnet for some reason
+                px.Kill();
+                px.WaitForExit();
+                Assert.True(px.HasExited);
+                Assert.Equal(137, px.ExitCode); // 137 means the process was killed
+            }
+        }
+
+        [Theory, InlineData("Safari"), InlineData("\"Google Chrome\"")]
+        [PlatformSpecific(TestPlatforms.OSX)]
+        // [OuterLoop("Opens browser")]
+        public void ProcessStart_OpenUrl_UsesSpecifiedApplication(string applicationToOpenWith)
+        {
+            using (var px = Process.Start("/usr/bin/open", "https://github.com/dotnet/corefx -a " + applicationToOpenWith))
+            {
+                Assert.False(px.HasExited);
+                px.WaitForExit();
+                Assert.True(px.HasExited);
+                Assert.Equal(0, px.ExitCode); // Exit Code 0 from open means success
+            }
+        }
+
+        [Theory, InlineData("-a Safari"), InlineData("-a \"Google Chrome\"")]
+        [PlatformSpecific(TestPlatforms.OSX)]
+        // [OuterLoop("Opens browser")]
+        public void ProcessStart_UseShellExecuteTrue_OpenUrl_SuccessfullyReadsArgument(string arguments)
+        {
+            var startInfo = new ProcessStartInfo { UseShellExecute = true, FileName = "https://github.com/dotnet/corefx", Arguments = arguments };
+            using (var px = Process.Start(startInfo))
+            {
+                Assert.NotNull(px);
+                // px.Kill(); // uncommenting this changes exit code to 137, meaning process was killed
+                px.WaitForExit();
+                Assert.True(px.HasExited);
+                Assert.Equal(0, px.ExitCode);
+            }
+        }
+
+        [Fact, PlatformSpecific(TestPlatforms.OSX)]
+        // [OuterLoop("Opens program")]
+        public void ProcessStart_UseShellExecuteTrue_TryOpenFileThatDoesntExist_ReturnsExitCode1()
```

Comment:
where is the Linux version of this test? also, why is this OSX specific? If it matches Windows behavior, it should be a test for all platforms?

Reply:
There is another test similar to this for Linux, but since the return code is different for linux I split it to two different tests. (the exit code is coming from xdg-open and open). 
I'll add a comment to explain why I split them to two tests